purchase_commercial_partner/models/purchase.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2016-2019 Akretion France (https://akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    commercial_partner_id = fields.Many2one(
+        'res.partner', related='partner_id.commercial_partner_id', store=True,
+        index=True, string='Commercial Vendor', readonly=True)
```

Comment:
@alexis-via no need to `readonly`

Reply:
I think yes, related fields with full access leads to weird and unwanted behavior, bad performance: crazy sql.

In master, related fields are now readonly by default, isn't it ?

Reply:
In v12, related are already `readonly` by default.

Reply:
you're right ;-)
https://github.com/odoo/odoo/blob/12.0/odoo/fields.py#L422

Reply:
We included in migration guide the reverse: if you rely on default `False` value (for example, for res.config.settings fields), you have to explicitly set the contrary: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0#tasks-to-do-in-the-migration

That's why the Python mantra "Explicit better than implicit" works. For this case, it's harmless to leave it explicit.